### PR TITLE
Us14 ref

### PR DESCRIPTION
**Pull request by Phemus**
Pull request adds a uuid to each created deed end modifies get deed endpoint to use the new ref

### CODE REVIEW

**Comment by tsangsimon** [comment]
:+1:

File: application/deed/model.py [hunk 1]
```diff
@@ -8,6 +8,7 @@ class Deed(db.Model):
     __tablename__ = 'deed'
 
     id = db.Column(db.Integer, primary_key=True)
+    uid = db.Column(db.String, nullable=False)
```
Comment (endamccormack) [inline application/deed/model.py]: uid is a very nondescript name for what it does

File: tests/test_app.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def test_get_endpoint(self, mock_query):
         mock_instance_response = mock_query.filter_by.return_value
         mock_instance_response.first.return_value = DeedModelMock()
 
-        response = self.app.get('/deed/1')
+        response = self.app.get('/deed/AB1234')
```
Comment (endamccormack) [inline tests/test_app.py]: Might just throw endpoint into a variable for reuse

**Comment by tsangsimon** [comment]
:+1:  ok with the changes there Enda

**Comment by Kirosoft** [comment]
Following discussion with Ramin & Simon: We need to double check that the unique reference is genuinely unique - stripping the first 6 characters from the uuid4() function could increase the chance of a hash collision. Suggest: chore to review the use of uuid4

**Comment by Kirosoft** [comment]
:+1:

**Comment by tsangsimon** [comment]
happy to have a tech chore out of this and move the current work on

**Pull request merged**